src/main/java/pl/cleankod/exchange/core/usecase/AccountServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+
+package pl.cleankod.exchange.core.usecase;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import pl.cleankod.exchange.core.domain.Account;
+import pl.cleankod.exchange.core.domain.Money;
+import pl.cleankod.exchange.core.gateway.AccountRepository;
+import pl.cleankod.exchange.core.gateway.AccountService;
+import pl.cleankod.exchange.core.gateway.CurrencyConversionService;
+import pl.cleankod.exchange.entrypoint.model.AccountNotFoundException;
+import pl.cleankod.exchange.entrypoint.model.AccountResponse;
+import pl.cleankod.exchange.entrypoint.model.CurrencyConversionException;
+
+import java.util.Currency;
+import java.util.Optional;
+
+@Service
+public class AccountServiceImpl implements AccountService {
```

Comment:
I noticed there is only a single implementation of the `AccountService` interface. Why did you extract an interface?